components/DarkModeSwitch.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { useState } from 'react';
+import Toggle from 'react-toggle';
+import useDarkMode from 'use-dark-mode';
+import SunIcon from './SunIcon';
+import MoonIcon from './MoonIcon';
+
+const DarkModeSwitch = () => {
+    const storageKey = 'darkMode';
+    let isDarkModeEnabled = true;
+    try {
+        isDarkModeEnabled = localStorage.getItem(storageKey) === 'true';
+    } catch (err) {}
```

Comment:
Maybe it's nice to recognise the system default and only overwrite if it's toggled again. This can be done via this line `window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches` that reads out the CSS media query.  The window is undefined needs to be filtered here.

```suggestion
const hasPersistedDarkMode: boolean = window.localStorage.getItem('DARK_MODE') !== undefined;
const isDarkModeEnabled = (hasPersistedDarkMode && localStorage.getItem('darkMode') === 'true') 
    || window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches 
    || false;
``` 

_Not the nicest code, but the flow would be something like:_
Check if dark mode is persisted  --> set persisted value --> else read system default.
The toggle will then always persist the toggled value.